pylint/checkers/refactoring/recommendation_checker.py

```diff
@@ -83,21 +83,22 @@ def _check_consider_iterating_dictionary(self, node: nodes.Call) -> None:
             return
         if node.func.attrname != "keys":
             return
+        comp_ancestor = utils.get_node_first_ancestor_of_type(node, (nodes.Compare,))
```

Comment:
```py
comp_ancestor = utils.get_node_first_ancestor_of_type(node, nodes.Compare)
```
This could then be simplified.

Reply:
Added to #5447 as a follow up issue to this.

I initially wanted to keep the typing of the function simple so thought it would be okay to only allow tuples, but we could allow both.